docs/lib/dynamic_config.md

```diff
@@ -0,0 +1,579 @@
+    - ```python
+      DATABASES = "@merge {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': 'db.sqlite3'}}"
+      ```
+      Converts to `DATABASES.update({'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': 'db.sqlite3'}})`  <br><br>
+    - ```python
+      PERSON = "@merge name=John"
+      ```
+      Converts to `PERSON.update({'name': 'John'})`<br><br>  
+
+- `@merge_unique` will merge the setting with the previous one, but only if the setting is not already present in the previous one.
+    - ```python
+      INSTALLED_APPS = "@merge_unique new_app_to_add_only_if_not_present_yet"
+      ```
+      Converts to `if "new_app" not in INSTALLED_APPS: INSTALLED_APPS.append("new_app")`  <br><br>
+
+- `@insert` will insert the setting in the previous one, if the setting is a list, the new setting will be inserted in the first position or the provided index.
+    - ```python
+      INSTALLED_APPS = "@insert new_app_to_add"
+      ```
+      Converts to `INSTALLED_APPS.insert(0, "new_app_to_add")`  <br><br>
+    - ```python
+      INSTALLED_APPS = "@insert 2 new_app_to_add"
+      ```
+      Converts to `INSTALLED_APPS.insert(2, "new_app_to_add")`  <br><br>
+
+#### Meta Values
+
+Besides the converters, Dynaconf also provides a set of meta values that can be used to define the intention of merging the settings in the value itself,
+metavalues can be added to lists as string and to dictionaries as keys.
+
+
+- `dynaconf_merge`
+    - ```python
+      INSTALLED_APPS = ["new_app_to_add", "dynaconf_merge"]
+      ```
+      Converts to `INSTALLED_APPS.append("new_app_to_add")`  <br><br>
+    - ```python
+      DATABASES = {"default": {"ENGINE": "django.db.backends.sqlite3", "NAME": "db.sqlite3", "dynaconf_merge": True}}
+      ```
+      Converts to `DATABASES.update({'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': 'db.sqlite3'}})`  <br><br>
+
+- `dynaconf_merge_unique`
+    - ```python
+      INSTALLED_APPS = ["new_app_to_add_only_if_not_present", "dynaconf_merge_unique"]
+      ```
+      Converts to `if "new_app" not in INSTALLED_APPS: INSTALLED_APPS.append("new_app")`  <br><br>
+
+
+The metavalue itself will be removed from the setting after the conversion.
+
+
+#### Deferred Post Hooks
+
+In some scenarions the value that needs to be set is dependent on other setting values that are not available yet, because dynaconf will have
+the final state only after resolving the load of all the files and environment variables, it is possible to use deferred post hooks to set the value after the final state is resolved.
+
+```python
+from dynaconf import post_hook
+
+@post_hook
+def add_ldap_only_if_flag_is_set(settings) -> dict:
+    data = {}  # this is the value to be merged
+    if settings.get("LDAP_ENABLED"): 
+        data["INSTALLED_APPS"] = "@merge_unique ldap_app"
+        data["LDAP_SERVER] = "ldap://server"
+        data["LDAP_PORT] = 389
+    return data
```

Comment:
Where would we set something like this? Is this in the settings.py file? Or in a django signals file?

Reply:
In any `.py` file that is loaded by dynaconf, it can be a `settings.py`, `defaults.py` or any file under `/etc/tower/conf.d/*py` in case of awx.

Dynaconf loads all variables and registers all hooks to be executed later.